File: java/src/LinearSearchList.java
package src;

public class LinearSearchList {

    // Search for an int in an array using linear search
    public static boolean linearSearch(int[] haystack, int needle) {
        for (int i = 0; i < haystack.length; ++i) {
            if (haystack[i] == needle) {
                return true;
            }
        }

        return false;
    }

}

File: typescript/src/LinearSearchList.ts
export default function linear_search(haystack: number[], needle: number): boolean {
    for (let i = 0; i < haystack.length; ++i) {
        if (haystack[i] === needle) {
            return true;
        }
    }

    return false;
}

// RUST
// fn linear_search(haystack: &[i32], needle: i32) -> bool {
//     for value in haystack {
//         if value == &needle {
//             return true;
//         }
//     }

//     return false;
// }